### ISSUE TITLE
Ledger mode account completion broken when auto-complete is on

### ISSUE THREAD

**Issue by lzmartinico** [syl20bnr/spacemacs#12024]
#### Description :octocat:
In ledger mode, cycling through accounts using the tab key when auto-complete is on is broken, by the addition of a space after each tab press. Sub accounts should not be separated by a space between the end of the word and the colon; if a space is added, completion of the subaccount is stopped. This is different from the behaviour of ledger mode in regular Emacs (or when auto-complete is off), where pressing the tab does not add a space, and thus if a subaccount is present, completion can continue. 

#### Reproduction guide :beetle:
- Start Emacs
- Turn auto-completion on
- Add new transaction (SPC m a)
- Start typing a subaccount
- Press tab to continue

*Observed behaviour:* :eyes: :broken_heart:
Account name is followed by a space

*Expected behaviour:* :heart: :smile:
Account name should not be followed by a space, and completion option for any subaccount should be available

#### System Info :computer:
- OS: darwin
- Emacs: 26.1
- Spacemacs: 0.200.13
- Spacemacs branch: master (rev. c7a103a77)
- Graphic display: nil
- Distribution: spacemacs
- Editing style: hybrid
- Completion: helm
- Layers:
```elisp
(auto-completion better-defaults bibtex emacs-lisp finance git helm latex markdown org python semantic
                 (shell :variables shell-default-height 30 shell-default-position 'bottom)
                 (spell-checking :variables enable-flyspell-auto-completion t)
                 syntax-checking version-control)
```
- System configuration features: NOTIFY ACL GNUTLS LIBXML2 ZLIB THREADS

#### Backtrace :paw_prints:
```
<<BACKTRACE IF RELEVANT>>
```

**Comment by Alexander-Miller** [comment]
Not caused by spacemacs:
https://github.com/company-mode/company-mode/issues/861
https://github.com/ledger/ledger-mode/issues/152

**Comment by lzmartinico** [comment]
Shouldn't Spacemacs disable company-mode by default then?

**Comment by larebsyed** [comment]
@lzmartinico I have been able to fix it by customizing the custom variable `pcomplete-termination-string` to empty string

**Comment by lzmartinico** [comment]
Thank you, that worked. Should this be added into the Spacemacs layer?

**Comment by larebsyed** [comment]
> Thank you, that worked. Should this be added into the Spacemacs layer?

@lzmartinico I am working on it, let see if it gets approved.

**Comment by lzmartinico** [comment]
Great! Feel free to add "fixes #12024" to your PR body so the issue is closed if it's approved

**Comment by duianto** [comment]
The PR #12077 has been cherry-picked into the develop branch.
